xbmc/addons/gui/GUIDialogAddonSettings.cpp

```diff
@@ -40,12 +47,49 @@ bool CGUIDialogAddonSettings::OnMessage(CGUIMessage& message)
   {
     case GUI_MSG_CLICKED:
     {
-      if (message.GetSenderId() == CONTROL_SETTINGS_CUSTOM_BUTTON)
+      const auto id = message.GetSenderId();
+      if (id == CONTROL_SETTINGS_CUSTOM_BUTTON)
       {
         OnResetSettings();
         return true;
       }
-      break;
+      else if (id == CONTROL_BTN_CONF_NAME)
+      {
+        const std::vector<uint32_t> ids = m_addon->GetKnownInstanceIds();
+        std::string instanceName = m_instanceName;
+        while (1)
+        {
+          bool ret = CGUIKeyboardFactory::ShowAndGetInput(instanceName, false);
```

Comment:
As I said in Slack, we should try to get rid of this whole process of entering a name for the instance before the settings dialog opens. Settings dialog itself should offer to edit the instance name.